x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/SnapshotRetentionConfiguration.java

```diff
@@ -214,14 +216,14 @@ public Integer getMaximumSnapshotCount() {
                         return false;
                     }
                 }
-
-                if (snapshotAge.compareTo(this.expireAfter) > 0) {
-                    logger.trace("[{}]: ELIGIBLE as snapshot age of {} is older than {}",
-                        snapName, snapshotAge.toHumanReadableString(3), this.expireAfter.toHumanReadableString(3));
+                final long snapshotAge = nowSupplier.getAsLong() - si.startTime();
+                if (snapshotAge > this.expireAfter.getMillis()) {
+                    logger.trace(() -> new ParameterizedMessage("[{}]: ELIGIBLE as snapshot age of [{}ms] is older than {}",
+                        snapName, snapshotAge, this.expireAfter.toHumanReadableString(3)));
```

Comment:
I would prefer to keep both, as having to do the calculation from milliseconds to real-time-a-human-can-interpret is annoying in the logs. So maybe:

```suggestion
                    logger.trace(() -> new ParameterizedMessage("[{}]: ELIGIBLE as snapshot age of [{}/{}ms] is older than {}",
                        snapName, new TimeValue(snapshotAge).toHumanReadableString(3), snapshotAge, this.expireAfter.toHumanReadableString(3)));
```